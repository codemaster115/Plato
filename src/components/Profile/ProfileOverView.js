import React from 'react';
import { Card, CardContent } from '@material-ui/core';
import styled from 'styled-components';
import { connect } from 'react-redux';

const ProfileCard = styled(CardContent)`
  display: grid;
  grid-template-columns: 50% 50%;
  grid-gap: 10px;
  grid-template-rows: 30px;
  height: 210px;
`;

const ProfileLeft = styled.div`
  display: grid;
  grid-template-columns: 90px 100px;
  grid-gap: 10px;
  grid-template-rows: 30px;
  height: 100%;
`;

const ProfileRight = styled.div`
  display: grid;
  grid-gap: 4px;
  grid-template-rows: 25px;
`;

const ProfileImage = styled.img`
  display: block;
  width: 90px;
  border-radius: 50%;
`;

const SubProfileImage = styled.img`
  display: block;
  width: 50px;
  margin: auto;
`;

const Image = styled.div`
  display: grid;
  grid-template-rows: 110px 60px;
  height: 100%;
`;

const Texts = styled.div`
  display: grid;
  grid-template-rows: 35px 20px 25px 25px 25px;
  gap: 10px;
  height: 100%;
`;

const Name = styled.div`
  box-sizing: border-box;
  height: 35px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 31px;
  font-weight: bold;
  color: rgb(51, 51, 51);
  line-height: 1;
`;

const Title = styled.div`
  box-sizing: border-box;
  height: 20px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 17px;
  font-weight: bold;
  color: rgb(51, 51, 51);
  line-height: 1;
`;

const Email = styled.div`
  box-sizing: border-box;
  height: 25px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112, 112, 112);
  line-height: 1.4;
  padding-top: 3.5px;
  word-break: break-word;
`;

const Icon = styled.button`
  box-sizing: border-box;
  height: 25px;
  width: 25px;
  text-align: center;
  cursor: pointer;
  background: none;
  border: none;
  color: rgba(69, 75, 128, 0.7);
  font-size: 25px;
  border-radius: 4px;
`;

const Status = styled.div`
  box-sizing: border-box;
  height: 25px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112, 112, 112);
  line-height: 1.4;
  padding-top: 3.5px;
`;

const TextIndex = styled.div`
  box-sizing: border-box;
  height: 25px;
  width: 411.418px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112, 112, 112);
  line-height: 1.4;
  padding-top: 3.5px;
`;

const Notes = styled.div`
  box-sizing: border-box;
  height: 25px;
  width: 411.418px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112, 112, 112);
  line-height: 1.4;
  padding-top: 32px;
`;

export class ProfileOverViewComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      mentee: [],
    };
  }

  UNSAFE_componentWillReceiveProps({ profileInfo }) {
    if (profileInfo) {
      this.setState({ mentee: profileInfo });
    }
  }

  render() {
    return (
      <Card>
        <ProfileCard>
          <ProfileLeft>
            <Image>
              <ProfileImage
                alt="profile"
                src={
                  this.state.mentee.picture
                    ? this.state.mentee.picture.thumbnailUrl
                    : ''
                }
              />
              <SubProfileImage
                alt="profile"
                src={
                  this.state.mentee.company
                    ? this.state.mentee.company.logo
                      ? this.state.mentee.company.logo.thumbnailUrl
                      : ''
                    : ''
                }
              />
            </Image>
            <Texts>
              <Name>{this.state.mentee.firstName}</Name>
              <Title>
                {this.state.mentee.title} at {this.state.mentee.companyName}
              </Title>
              <Email>{this.state.mentee.email}</Email>
              <Icon className="fa fa-linkedin-square inner-element bubble-element clickable-element" />
              <Status>Status: {this.state.mentee.obStatus}</Status>
            </Texts>
          </ProfileLeft>
          <ProfileRight>
            <TextIndex>
              <strong>Team size: </strong>
              {this.state.mentee.engineeringTeamSize}
            </TextIndex>
            <TextIndex>
              <strong>Years of management experience: </strong>
              {this.state.mentee.yearsInManagementNb}
            </TextIndex>
            <TextIndex>
              <strong>Number of employees in the company: </strong>
              {this.state.mentee
                ? this.state.mentee.company
                  ? this.state.mentee.company.nbEmployees
                  : ''
                : ''}
            </TextIndex>
            <Notes>
              <strong>Notes: </strong>
            </Notes>
          </ProfileRight>
        </ProfileCard>
      </Card>
    );
  }
}

const mapStateToProps = state => ({
  profileInfo: state.Profile.profileInfo,
});

const mapDispatchToProps = {};

ProfileOverViewComponent.propTypes = {};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ProfileOverViewComponent);
