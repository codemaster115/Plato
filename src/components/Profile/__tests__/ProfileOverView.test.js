import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { ProfileOverViewComponent } from '../ProfileOverView';

describe('<ProfileOverViewComponent/>', () => {
  it('Compare snapshots', () => {
    const Component = renderer.create(<ProfileOverViewComponent />);

    const Tree = Component.toJSON();
    expect(Tree).toMatchSnapshot();
  });

  it('Renders without crashing', () => {
    mount(<ProfileOverViewComponent />);
  });
});
