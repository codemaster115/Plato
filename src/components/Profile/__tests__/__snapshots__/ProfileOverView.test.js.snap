// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileOverViewComponent/> Compare snapshots 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  grid-gap: 10px;
  grid-template-rows: 30px;
  height: 210px;
}

.c1 {
  display: grid;
  grid-template-columns: 90px 100px;
  grid-gap: 10px;
  grid-template-rows: 30px;
  height: 100%;
}

.c11 {
  display: grid;
  grid-gap: 4px;
  grid-template-rows: 25px;
}

.c3 {
  display: block;
  width: 90px;
  border-radius: 50%;
}

.c4 {
  display: block;
  width: 50px;
  margin: auto;
}

.c2 {
  display: grid;
  grid-template-rows: 110px 60px;
  height: 100%;
}

.c5 {
  display: grid;
  grid-template-rows: 35px 20px 25px 25px 25px;
  gap: 10px;
  height: 100%;
}

.c6 {
  box-sizing: border-box;
  height: 35px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 31px;
  font-weight: bold;
  color: rgb(51,51,51);
  line-height: 1;
}

.c7 {
  box-sizing: border-box;
  height: 20px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 17px;
  font-weight: bold;
  color: rgb(51,51,51);
  line-height: 1;
}

.c8 {
  box-sizing: border-box;
  height: 25px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112,112,112);
  line-height: 1.4;
  padding-top: 3.5px;
  word-break: break-word;
}

.c9 {
  box-sizing: border-box;
  height: 25px;
  width: 25px;
  text-align: center;
  cursor: pointer;
  background: none;
  border: none;
  color: rgba(69,75,128,0.7);
  font-size: 25px;
  border-radius: 4px;
}

.c10 {
  box-sizing: border-box;
  height: 25px;
  width: 231.922px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112,112,112);
  line-height: 1.4;
  padding-top: 3.5px;
}

.c12 {
  box-sizing: border-box;
  height: 25px;
  width: 411.418px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112,112,112);
  line-height: 1.4;
  padding-top: 3.5px;
}

.c13 {
  box-sizing: border-box;
  height: 25px;
  width: 411.418px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112,112,112);
  line-height: 1.4;
  padding-top: 32px;
}

<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="MuiCardContent-root-29 c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <img
          alt="profile"
          className="c3"
          src=""
        />
        <img
          alt="profile"
          className="c4"
          src=""
        />
      </div>
      <div
        className="c5"
      >
        <div
          className="c6"
        />
        <div
          className="c7"
        >
           at 
        </div>
        <div
          className="c8"
        />
        <button
          className="fa fa-linkedin-square inner-element bubble-element clickable-element c9"
        />
        <div
          className="c10"
        >
          Status: 
        </div>
      </div>
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      >
        <strong>
          Team size: 
        </strong>
      </div>
      <div
        className="c12"
      >
        <strong>
          Years of management experience: 
        </strong>
      </div>
      <div
        className="c12"
      >
        <strong>
          Number of employees in the company: 
        </strong>
        
      </div>
      <div
        className="c13"
      >
        <strong>
          Notes: 
        </strong>
      </div>
    </div>
  </div>
</div>
`;
